src/huggingface_hub/hf_file_system.py

```diff
@@ -649,6 +654,58 @@ def _upload_chunk(self, final: bool = False) -> None:
             )
 
 
+class HfFileSystemStreamFile(fsspec.spec.AbstractBufferedFile):
+    def __init__(
+        self, fs: HfFileSystem, path: str, revision: Optional[str] = None, block_size=0, cache_type="none", **kwargs
+    ):
+        if block_size != 0:
+            raise ValueError(f"HfFileSystemStreamFile only supports block_size=0 but got {block_size}")
+        if cache_type != "none":
+            raise ValueError(f"HfFileSystemStreamFile only supports cache_type='none' but got {cache_type}")
+        try:
+            self.resolved_path = fs.resolve_path(path, revision=revision)
+        except FileNotFoundError as e:
+            if "w" in kwargs.get("mode", ""):
+                raise FileNotFoundError(
+                    f"{e}.\nMake sure the repository and revision exist before writing data."
+                ) from e
+        self.details = {"name": self.resolved_path.unresolve(), "size": None}
+        super().__init__(fs, self.resolved_path.unresolve(), block_size=block_size, cache_type=cache_type, **kwargs)
+        self.r: Optional[Response] = None
+        self.fs: HfFileSystem
+
+    def seek(self, loc, whence=0):
+        if loc == 0 and whence == 1:
+            return
+        if loc == self.loc and whence == 0:
+            return
+        raise ValueError("Cannot seek streaming HF file")
```

Comment:
```suggestion
        raise ValueError("Cannot seek streaming HF file.")
```

Reply:
Out of curiosity, it is that we don't support it (not implemented) or that it's not possible to do it?

Reply:
It's not possible to do random access with a streaming file from `requests`. If users need random access they should not load a streaming file (block_size=0). The fsspec HTTPFileSystem has the same behavior